manifests/client.pp

```diff
@@ -92,4 +95,13 @@
     hasrestart => true,
     require    => File['bacula-fd.conf'],
   }
+
+  if $pki_encryption {
+    Exec { 'create_keypair':
+      command => "/usr/bin/openssl genrsa -out private.key 4096 && /usr/bin/openssl req -new -key private.key -x509 -out public.crt -subj '/C=XX/ST=unknown/L=puppet-bacula/O=Bacula Backup/OU=backup/CN=${::fqdn}' && cat private.key public.crt >$pki_keypair && rm private.key public.crt",
```

Comment:
This exec should also be controlled by a $manage_pki_keypair parameter to allow users to provide custom keypair data through hiera or other means.  For example, our environment makes use of the puppet certificates signed by the puppet CA to enable TLS connections in bacula.

The default value would be set to true however users could override the value as needed.

Reply:
You are right. I haven't thought of that, although the exec would have only been triggered if the file would not be existent. So if you provide the file through another file resource for example the exec should not have been executed. Or if it would, your file resource would have overwritten the created keypair.